src/DAMA/DoctrineTestBundle/PHPUnit/PHPUnitListener.php

```diff
@@ -4,7 +4,29 @@
 
 use DAMA\DoctrineTestBundle\Doctrine\DBAL\StaticDriver;
 
-if (!class_exists('\PHPUnit_Framework_BaseTestListener')) {
+if (trait_exists('\PHPUnit\Framework\TestListenerDefaultImplementation') && PHP_VERSION_ID >= 70100) {
```

Comment:
do we need `&& PHP_VERSION_ID >= 70100` then? PHPUnit 7 requires PHP 7.1, right?

Reply:
See https://travis-ci.org/dmaicher/doctrine-test-bundle/jobs/336545621#L605 for error without such control

Reply:
Leading slash in name is invalid.
And 7.1 check is useless since TestListenerDefaultImplementation exists only in PHPUnit that requires 7.1+.

Reply:
Btw referenced Travis job uses PHP 7.0.25 which is invalid.

Reply:
Seems in the failing test the if here passes even with PHPUnit 6?

Maybe we can do this instead?

```diff
diff --git a/src/DAMA/DoctrineTestBundle/PHPUnit/PHPUnitListener.php b/src/DAMA/DoctrineTestBundle/PHPUnit/PHPUnitListener.php
index d28a5e9..29b2b05 100644
--- a/src/DAMA/DoctrineTestBundle/PHPUnit/PHPUnitListener.php
+++ b/src/DAMA/DoctrineTestBundle/PHPUnit/PHPUnitListener.php
@@ -4,7 +4,26 @@ namespace DAMA\DoctrineTestBundle\PHPUnit;
 
 use DAMA\DoctrineTestBundle\Doctrine\DBAL\StaticDriver;
 
-if (!class_exists('\PHPUnit_Framework_BaseTestListener')) {
+if (class_exists('\PHPUnit_Framework_BaseTestListener')) {
+    // PHPUnit 5
+    class PHPUnitListener extends \PHPUnit_Framework_BaseTestListener
+    {
+        public function startTest(\PHPUnit_Framework_Test $test)
+        {
+            StaticDriver::beginTransaction();
+        }
+
+        public function endTest(\PHPUnit_Framework_Test $test, $time)
+        {
+            StaticDriver::rollBack();
+        }
+
+        public function startTestSuite(\PHPUnit_Framework_TestSuite $suite)
+        {
+            StaticDriver::setKeepStaticConnections(true);
+        }
+    }
+} elseif (class_exists('\PHPUnit\Framework\BaseTestListener')) {
     // PHPUnit 6+
     class PHPUnitListener extends \PHPUnit\Framework\BaseTestListener
     {
@@ -24,20 +43,22 @@ if (!class_exists('\PHPUnit_Framework_BaseTestListener')) {
         }
     }
 } else {
-    // PHPUnit 5
-    class PHPUnitListener extends \PHPUnit_Framework_BaseTestListener
+    // PHPUnit 7+
+    class PHPUnitListener implements \PHPUnit\Framework\TestListener
     {
-        public function startTest(\PHPUnit_Framework_Test $test)
+        use \PHPUnit\Framework\TestListenerDefaultImplementation;
+
+        public function startTest(\PHPUnit\Framework\Test $test): void
         {
             StaticDriver::beginTransaction();
         }
 
-        public function endTest(\PHPUnit_Framework_Test $test, $time)
+        public function endTest(\PHPUnit\Framework\Test $test, $time): void
         {
             StaticDriver::rollBack();
         }
-
-        public function startTestSuite(\PHPUnit_Framework_TestSuite $suite)
+        
+        public function startTestSuite(\PHPUnit\Framework\TestSuite $suite): void
         {
             StaticDriver::setKeepStaticConnections(true);
         }

Reply:
It exists as of 6.4.0: https://github.com/sebastianbergmann/phpunit/commit/9d5924f4bf4d4f41529da5e2c98c1d8ed9495e00#diff-f014f0750e1e6f18c6598f8c01d788a5

Reply:
@Majkl578 yeah exactly so we cannot use it to distinguish versions 6 and 7

Reply:
As it looks you can't properly keep support for PHPUnit 7 and PHP <7.1 at the same time.
I'd suggest to drop <7.1 and release new minor/major. Supporting anything below 7.1 is obsolelte anyway nowadays...

Reply:
Well, currently this PR is working with both: phpunit7 on php7.1 and phpunit6 on php 7.0
Anyway, I agree that dropping php7.0 would be good at this point, since it's already out of support.
What do you think @dmaicher ? I can adapt this PR to drop php7.0

Reply:
Legacy users don't need shiny new versions and features anyway. 👍 for drop.

Reply:
:+1:  for also dropping php 7.0 support

Reply:
I will also drop phpunit 5 support completely in a separate PR as its not officially supported anymore as of today